icu4j/main/core/src/main/java/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -440,6 +441,22 @@ private static void getCurrencyLongNameData(ULocale locale, Currency currency, S
         // In ICU4J, this method gets a CurrencyData from CurrencyData.provider.
         // TODO(ICU4J): Implement this without going through CurrencyData, like in ICU4C?
         Map<String, String> data = CurrencyData.provider.getInstance(locale, true).getUnitPatterns();
+    	// Entries in the data map are filled in with any CurrencyUnitPatterns data for locale,
+    	// or if there is no CurrencyUnitPatterns data for locale since the patterns all inherited
+    	// from the "other" pattern in root (which is true for many locales in CLDR 46), then only
+    	// the "other" entry has a currency pattern. So now what we do is: For all valid plural keywords
+    	// for the locale, if the corresponding outArray[] entry is bogus, fill it in from the "other"
+    	// entry. 
+        PluralRules prules = PluralRules.forLocale(locale);
+        Set<String> keywords = prules.getKeywords();
+        String otherPattern = data.get("other");
+        if (keywords != null && otherPattern!= null) {
+            for (String keyword: keywords) {
+                if (!keyword.equals("other") && !data.containsKey(keyword)) {
+                    data.put(keyword, otherPattern);
+                }
+            }
+        } 
```

Comment:
I'm still not sure what's going on here, but this code looks different from (and shorter than) the corresponding code in ICU4C above.  Are they really doing the same thing?

Reply:
It is really the same. The difference is, ICU4C is using an array, which has the bogus entries for the values that are not present. Those bogus entries are causing ICU4C to bail out. In ICU4J, a map is used instead of an array, so instead of updating an entry at an index, we just add a new entry with the key of "one" "two" "few" etc

Reply:
Okay, I'll take your word for it.